syntax = "proto3";
package steward;

import "steward/asset.proto";
import "steward/schedule.proto";

message Maintenance {
  string _id = 1;
  string name = 2;
  string description = 3;
  Asset asset = 4;
  bool enabled = 5;
  Schedule schedule = 6;
}

message GetMaintenanceRequest {
  string _id = 1;
}

message DeleteMaintenanceRequest {
  string _id = 1;
}

message UpdateMaintenanceRequest {
  string _id = 1;
  Maintenance maintenance = 2;
}

message ListMaintenancesRequest {
  Asset asset = 2;
}
